lmdeploy/messages.py

```diff
@@ -298,5 +298,7 @@ class VisionConfig:
         max_batch_size (int): the max image size passed to the model, since
             some models will use image patch, the actual running batch could
             be larger than this value.
+        thread_safe (bool): thread safe engine instance.
     """
     max_batch_size: int = 1
+    thread_safe: bool = False
```

Comment:
When should we set it `True`?

Reply:
If you want to use `pipeline.__call__`, `pipeline.chat`, `pipeline.stream_infer` in multi-thread.

Reply:
Let's clarify it in the docstring
```
thread_safe (bool): Specifies whether the engine instance is thread-safe. Please set it to True when using the pipeline in a multi-threaded environment.
```